### PR TITLE
Supplements of guideline of compilation on macOS with make and clean up some code

### PR DESCRIPTION
**Pull request by LykxSassinator**
I have tried to compile the KVRocks on MacOS, with 11.4 Big Sur, finding that there have missed some tools for compilation on the first time. So i have make the supplements in the README.md.
Unfortunately, it seems that there still exists some limitations and errs while compiling KVRocks on MacOS by make with scripts automatically, which should be manually interfered with. Therefore, currently, it's more recommended to use cmake on MacOS or compiling it on other Linux systems.

### CODE REVIEW

**Comment by LykxSassinator** [comment]
It's my unpleasable push of two separatable commits. The second commit is used for making several tiny improvements by converting unnecessary usages of Copy Constructor with the appropriate movement ops.

**Comment by git-hulk** [comment]
cool, thanks for your contribution. Can u help to fix issues found by cpplint?

```
src/redis_db.cc:468:  Extra space for operator  ++;  [whitespace/operators] [4]
src/redis_hash.cc:178:  Add #include <utility> for move  [build/include_what_you_use] [4]
```

File: src/redis_db.cc [hunk 1]
```diff
@@ -383,10 +386,8 @@ rocksdb::Status Database::FindKeyRangeWithPrefix(const std::string &prefix,
   // it's ok to increase the last char in prefix as the boundary of the prefix
   // while we limit the namespace last char shouldn't be larger than 128.
   std::string next_prefix = prefix;
-  char last_char = next_prefix.back();
-  last_char++;
-  next_prefix.pop_back();
-  next_prefix.push_back(last_char);
+  auto prefix_size = prefix.size();
+  next_prefix[prefix_size - 1] ++;
```
Comment (ShooterIT) [inline src/redis_db.cc]:
```suggestion
  next_prefix[prefix_size - 1]++;
```

File: src/cron.cc [hunk 2]
```diff
@@ -29,7 +29,7 @@ Status Cron::SetScheduleTime(const std::vector<std::string> &args) {
     if (!s.IsOK()) {
       return Status(Status::NotOK, "time expression format error : " + s.Msg());
     }
-    new_schedulers.push_back(st);
+    new_schedulers.emplace_back(std::move(st));
```
Comment (ShooterIT) [inline src/cron.cc]: i think it is not a good idea to call move, since `st` may be access again

Comment (ShooterIT) [reply src/cron.cc]: actually, I also did that in parsing requests, but i find c++ standard doesn't recommend that, i just think we don't call that frequently.

Comment (ShooterIT) [reply src/cron.cc]: And i find we don't define its `move constructor`, so this change doesn't take any improvements. And all members of schedule just are `int` type.

File: src/redis_db.cc (hunk 1 above)
Comment (LykxSassinator) [reply src/redis_db.cc]: It's my pleasure. Thx for the contributes of @ShooterIT .

File: src/cron.cc (hunk 2 above)
Comment (LykxSassinator) [reply src/cron.cc]: Yep, at this point, std::move was not a recommendable operation for the user-defined `st` object. I would revoke it here.

**Review by LykxSassinator — COMMENTED** [review]
All codes have been reviewed and checked again.
And It's my pleasure to make contributions to KVRocks.

File: README.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ sudo yum install -y epel-release && sudo yum install -y git gcc gcc-c++ make sna
 sudo apt-get install gcc g++ make libsnappy-dev autoconf automake libtool which libgtest-dev
 
 # MACOSX
-brew install snappy googletest
+brew install autoconf automake libtool snappy googletest
```
Comment (ShooterIT) [inline README.md:50]: do we need to brew install gcc? I find we install gcc in CI for macOS.
https://github.com/KvrocksLabs/kvrocks/blob/unstable/.github/workflows/kvrocks.yaml#L60

**Comment by ShooterIT** [comment]
cpplint error https://github.com/KvrocksLabs/kvrocks/runs/3101400846#step:4:721 please add necessary header files.

File: README.md (hunk 3 above)
Comment (LykxSassinator) [reply README.md:50]: The original version of GCC in MacOS is `4.2.1`. Although the compilation requirement requires a higher version(gcc >= 4.8), the compilation works around in fact. So, should we recommend the MacOS users upgrade the GCC version?

**Review by LykxSassinator — COMMENTED** [review]
Done.

**Review by LykxSassinator — COMMENTED** [review]
The error checked by cpplint has been fixed.

File: README.md (hunk 3 above)
Comment (ShooterIT) [reply README.md:50]: it seems that gcc is clang on macOS platform, it is fine for compilation.

**Review by ShooterIT — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]
LGTM

**Pull request merged**